### PR TITLE
Rules 2023 01

### LINKED ISSUE #585: COM DLL
## Features

- export: DllGetClassObject
- (export: DllRegisterServer)

## Additional context
https://docs.microsoft.com/en-us/windows/win32/api/combaseapi/nf-combaseapi-dllgetclassobject

### PR DESCRIPTION
**Pull request by mr-tz**
closes #585

### CODE REVIEW

File: host-interaction/wmi/connect-to-wmi-namespace-via-wbemlocator.yml [hunk 1]
```diff
@@ -9,15 +9,22 @@ rule:
...
           - bytes: 11 F8 90 45 3A 1D D0 11 89 1F 00 AA 00 4B 2E 24 = CLSID_WbemLocator
           - bytes: 87 A6 12 DC 7F 73 CF 11 88 4D 00 AA 00 4B 2E 24 = IID_IWbemLocator
-      - offset: 0x18 = ppv->ConnectServer
+      - or:
+        - and:
+          - arch: i386
+          - offset: 0xC = ppv->ConnectServer
+        - and:
+          - arch: amd64
+          - offset: 0x18 = ppv->ConnectServer
```
Comment (mike-hunhoff) [inline host-interaction/wmi/connect-to-wmi-namespace-via-wbemlocator.yml:27]: nice!

File: nursery/linked-against-libsodium.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+rule:
+  meta:
+    name: linked against libsodium
+    namespace: linking/static/crypto
+    authors:
+      - "@mr-tz"
+    scope: file
+    mbc:
+      - Cryptography::Crypto Library [C0059]
```
Comment (mike-hunhoff) [inline nursery/linked-against-libsodium.yml:9]: It would be useful to add a `description` meta with a short libsodium description.

**Review by mike-hunhoff — APPROVED** [review]
Great, thank you! I left one small suggestion via comment but not a blocker.

File: nursery/linked-against-libsodium.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+rule:
+  meta:
+    name: linked against libsodium
+    namespace: linking/static/crypto
+    authors:
+      - "@mr-tz"
```
Comment (mr-tz) [inline nursery/linked-against-libsodium.yml:6]:
```suggestion
      - "@mr-tz"
    description: Sodium is a software library for encryption, decryption, signatures, password hashing and more.
```

File: nursery/implement-com-dll.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+rule:
+  meta:
+    name: implement COM DLL
```
Comment (williballenthin) [inline nursery/implement-com-dll.yml:3]: discussion: is this a COM server or a COM DLL?

**Review by williballenthin — APPROVED** [review]

**Pull request merged**